src/data_gradients/feature_extractors/common/image_duplicates.py

```diff
@@ -0,0 +1,264 @@
+             valid_image_dir=/data/coco/images/val2017/):
+
+            self.train_dups: [['/data/coco/images/train2017/000000216265.jpg', '/data/coco/images/train2017/000000411274.jpg']...]
+            self.valid_dups: [] -> no duplicates in validation
+            self.intersection_dups: [['/data/coco/images/train2017/000000080010.jpg', '/data/coco/images/val2017/000000140556.jpg'],
+                                        ['/data/coco/images/train2017/000000535889.jpg', '/data/coco/images/val2017/000000465129.jpg']]
+
+            self.train_dups_appearences: 72
+            self.validation_dups_appearences: 0
+            self.intersection_train_appearnces: 2
+            self.intersection_val_appearnces: 2
+
+            IMPORTANT: We get len(self_train_dups) = 35, but there are 72 appearences pf duplicated images in the train directory.
+                This is because we have two triplet duplicates inside our train data.
+
+
+    NOTES:
+         - DOES NOT TAKE IN ACCOUNT ANY DATASET INTERNAL LOGIC, SIMPLY EXTRACTS THE DUPLICATES FROM THESE DIRECTORIES.
+         - If an image in the image directory can't be loaded, no duplicates are searched for the image.
+         - Supported image formats: 'JPEG', 'PNG', 'BMP', 'MPO', 'PPM', 'TIFF', 'GIF', 'SVG', 'PGM', 'PBM', 'WEBP'.
+    """
+
+    def __init__(self, train_image_dir: str = None, valid_image_dir: str = None):
+        """
+        :param train_image_dir: str = None, The directory containing all train images. When None, will ask the user
+            using prompt for input.
+
+        :param valid_image_dir: str = None. Ignored when val_data of the AbstractManager is None. The directory containing all
+            valid images. When None, will ask the user using prompt for input.
+        """
+        super().__init__()
+        self.train_dups = None
+        self.valid_dups = None
+        self.intersection_dups = None
+        self.train_image_dir = train_image_dir
+        self.valid_image_dir = valid_image_dir
+
+    def prep_for_duplicated_detection(self, train_data: Iterable, val_data: Iterable):
+        """
+        Called in AbstractManager.__init__
+        In case train_image_dir: str = None, valid_image_dir: str = None in __init__, will ask the user to pass them.
+
+        train_data: Iterable, the train_data used by AbstractManager (not used at the moment, but acts as a placeholder that
+            will later be used to derrive the diretory automatically according to its type - i.e ImageFolder.root etc).
+
+        val_data, Iterbale, the val_Data used by AbstractManager.
+        """
+        # TODO: ADD AUTOMATIC EXTRACTION FOR SG DATASETS
+        if self.train_image_dir is None:
+            self.train_image_dir = self._get_image_dir("train")
+        if val_data is not None and self.valid_image_dir is None:
+            self.valid_image_dir = self._get_image_dir("validation")
```

Comment:
`train_data` and `val_data` are not used ?
Is it because you changed your approach and eventually wen for working with folders directly ?

Reply:
please read the docs, I put the effort :)